### PR TITLE
fixed an LMDB issue on OpenBSD (#2699) which leads to a crash in monerod

### PR DESCRIPTION
**Pull request by ston1th**

### CODE REVIEW

File: src/blockchain_db/lmdb/db_lmdb.cpp [hunk 1]
```diff
@@ -1123,6 +1123,9 @@ void BlockchainLMDB::open(const std::string& filename, const int db_flags)
 
   m_folder = filename;
 
+#ifdef __OpenBSD__
+  mdb_flags |= MDB_WRITEMAP;
```
Comment (moneromooo-monero) [inline src/blockchain_db/lmdb/db_lmdb.cpp]: Would you mind adding sometnhing like:
if ((mdb_flags & MDB_WRITEMAP) == 0) MCLOG_RED(level, "global", "Running on OpenBSD: forcing WRITEMAP");
before the |=
Always helps to see what's going on when we get logs.

Comment (ston1th) [reply src/blockchain_db/lmdb/db_lmdb.cpp]: Ohh good idea! Sorry diddn't thought of something like this..

This appears in the log:
`2017-10-22 11:54:01.441    0x109f7cae6280        INFO    global  src/blockchain_db/lmdb/db_lmdb.cpp:1122 Running on OpenBSD: forcing WRITEMAP`

**Comment by moneromooo-monero** [comment]
Oh, I just realized you have that split over three patches, can you squash those please.

**Comment by moneromooo-monero** [comment]
Sorry to nitpick again, but could you make the commit message explanatory please ?

**Comment by fluffypony** [comment]
@ston1th congrats on your first merged pull request!!

thumbs up

**Pull request merged**